### PR TITLE
Add `LoggerFactory` docs

### PR DESCRIPTION
**Pull request by danicheg**
These docs are based on work made by @lorandszakacs in https://github.com/typelevel/log4cats/pull/421. So all credits go to them.
This covers a little part of the `LoggerFactory`, a basic functionality at most. It'd be great to complete it subsequently.

### CODE REVIEW

**Comment by danicheg** [comment]
I will be pretty happy if @hamnis and @lorandszakacs can have to look at this.

**Comment by hamnis** [comment]
Great stuff!

File: README.md [hunk 1]
```diff
@@ -79,6 +79,54 @@ def log[F[_]: Sync: Logger] =
   } yield ()
 ```
 
+## Logging using capabilities
+
+You can work with logging using capabilities. It's implemented via the `LoggerFactory` trait.
+You instantiate it once in your application (dependent on the specific logging backend you use)
```
Comment (FunFunFine) [inline README.md:85]:
```suggestion
You instantiate it once (dependent on the specific logging backend you use)
```
as the part `in your application` doubles in sentence.

File: README.md [hunk 2]
```diff
@@ -79,6 +79,54 @@ def log[F[_]: Sync: Logger] =
...
+and pass this around in your application.
+
+This brings several advantages:
+
+* it's no more need to pass everywhere the very powerful `F[_]: Sync` constraint
```
Comment (FunFunFine) [inline README.md:90]:
```suggestion
* it's no more needed to pass the very powerful `F[_]: Sync` constraint everywhere
```
some grammar

File: README.md [hunk 3]
```diff
@@ -79,6 +79,54 @@ def log[F[_]: Sync: Logger] =
...
+  that can do almost anything when you only need logging.
+* you have control of loggers creation, and you can even add in whatever custom
+  functionality you need for your applications here. E.g. create loggers that also push logs
+  to some external providers by giving a custom implementation of this trait.
```
Comment (FunFunFine) [inline README.md:94]:
```suggestion
* you have control of loggers creation, and you can even add in whatever custom
  functionality you need for your applications here by giving a custom implementation of this trait. For example, create loggers that also push logs to some external providers.
```
describe the way before talking about examples

File: README.md [hunk 4]
```diff
@@ -79,6 +79,54 @@ def log[F[_]: Sync: Logger] =
...
+
+If you are unsure how to create a `LoggerFactory[F]`, then you can to look at the `log4cats-slf4j`,
```
Comment (FunFunFine) [inline README.md:96]:
```suggestion
If you are unsure how to create a `LoggerFactory[F]`, then you can look at the `log4cats-slf4j`,
```

File: README.md [hunk 5]
```diff
@@ -79,6 +79,54 @@ def log[F[_]: Sync: Logger] =
...
+// create our LoggerFactory
+implicit val logging: LoggerFactory[IO] = Slf4jFactory[IO]
+
+// we summon LoggerFactory instance, and create logger
+val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+logger.info("logging in IO!"): IO[Unit]
+
+def useLogging[F[_]: LoggerFactory] = 
+  LoggerFactory[F].getLogger.info("yay! effect polymorphic code")
+useLogging[IO]
```
Comment (FunFunFine) [inline README.md:127]:
```suggestion
// basic example of a service using LoggerFactory
class LoggerUsingService[F[_]: LoggerFactory] {
  val logger = LoggerFactory[F].getLogger
  def use(args: String): F[Unit] = logger.info("yay! effect polymorphic code") >> logger.debug(s"and $args")
}

new LoggerUsingService[IO].use
```
let's make this example closer to real world usage

File: docs/index.md [hunk 6]
```diff
@@ -83,6 +83,54 @@ def log[F[_]: Sync: Logger] =
...
+// create our LoggerFactory
+implicit val logging: LoggerFactory[IO] = Slf4jFactory[IO]
+
+// we summon LoggerFactory instance, and create logger
+val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+logger.info("logging in IO!"): IO[Unit]
+
+def useLogging[F[_]: LoggerFactory] = 
+  LoggerFactory[F].getLogger.info("yay! effect polymorphic code")
+useLogging[IO]
```
Comment (FunFunFine) [inline docs/index.md:131]:
```suggestion
// basic example of a service using LoggerFactory
class LoggerUsingService[F[_]: LoggerFactory] {
  val logger = LoggerFactory[F].getLogger
  def use(args: String): F[Unit] = logger.info("yay! effect polymorphic code") >> logger.debug(s"and $args")
}

new LoggerUsingService[IO].use
```
let's make this example closer to real-world usage x2

**Comment by lorandszakacs** [comment]
Looks good, I'd incorporate @FunFunFine's suggestions though

**Comment by danicheg** [comment]
I've tackled most of the comments, so probably this PR is quite ok to get merged. If anyone wants to continue improving these docs, don't be shy and open a PR.

**Pull request merged**